src/manage_sql_secinfo.c

```diff
@@ -2793,18 +3273,31 @@ update_scap_cves ()
                          GINT_TO_POINTER (iterator_int (&cpes, 1)));
 
   count = 0;
-  while ((xml_path = g_dir_read_name (dir)))
-    if (fnmatch ("nvdcve-2.0-*.xml", xml_path, 0) == 0)
-      {
-        if (update_cve_xml (xml_path, hashed_cpes))
-          {
-            g_dir_close (dir);
-            g_hash_table_destroy (hashed_cpes);
-            cleanup_iterator (&cpes);
-            return -1;
-          }
-        count++;
-      }
+  while ((cve_path = g_dir_read_name (dir)))
+    {
+      if (fnmatch ("nvdcve-1.1-*.json", cve_path, 0) == 0)
+        {
+          if (update_cve_json (cve_path, hashed_cpes))
+            {
+              g_dir_close (dir);
+              g_hash_table_destroy (hashed_cpes);
+              cleanup_iterator (&cpes);
+              return -1;
+            }
+          count++;
+        }
+      if (fnmatch ("nvdcve-2.0-*.xml", cve_path, 0) == 0)
+        {
+          if (update_cve_xml (cve_path, hashed_cpes))
+            {
+              g_dir_close (dir);
+              g_hash_table_destroy (hashed_cpes);
+              cleanup_iterator (&cpes);
+              return -1;
```

Comment:
Maybe this needs to be an if-else to check for JSON first and if not fallback to XML ?

Reply:
I think if the directory has both JSON and XML files, then it will match them both. First will match the JSON then the XML. Right ? Do we need to handle this case ?

Reply:
I changed the last if to an else if, so it should be okay now, shouldn't it?